src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/SocketsHttpHandler.cs

```diff
@@ -213,5 +213,26 @@ public Func<SocketsHttpPlaintextStreamFilterContext, CancellationToken, ValueTas
             get => throw new PlatformNotSupportedException();
             set => throw new PlatformNotSupportedException();
         }
+
+        public static bool SupportsAutomaticDecompression
```

Comment:
Why do we have these differences from upstream?

Reply:
wasi doesn't compile without them, let me check why

Reply:
Reverted and remove some workarounds that were pending this merge